opal/util/proc.h

```diff
@@ -171,6 +171,6 @@ OPAL_DECLSPEC extern struct opal_proc_t *(*opal_proc_for_name)(const opal_proces
  * our own. This is to be used by all BTLs so we don't retrieve hostnames
  * unless needed. The returned value MUST NOT be free'd as it is
  * owned by the proc_t */
-OPAL_DECLSPEC char *opal_get_proc_hostname(const opal_proc_t *proc);
+OPAL_DECLSPEC extern char *(*opal_get_proc_hostname)(const opal_proc_t *proc);
```

Comment:
it feels like the proc code shouldn't be in the util library.  I'm guessing there's a dependency I'm missing, but that would certainly be my preference on a split point.

Reply:
I tried that, but there is no easy way because util being used via SUBDIRS, it generates a convenience library. If we want to split this, we will have to generate 2 convenience libraries, and include them accordingly at the upper level.

Reply:
I was originally leaning that way (splitting the util directory).  I'm out of time for a bit, but tonight I'll try to see if we can get rid of the subdirs for the util directory.  It's either because of the lex file (harder to get rid of) or the LTDLINCL, which shouldn't be needed at this point.